service/awsconfig/v5/resource/endpoints/resource.go

```diff
@@ -46,6 +37,9 @@ type Resource struct {
 // New creates a new configured endpoints resource.
 func New(config Config) (*Resource, error) {
 	// Dependencies.
+	if reflect.DeepEqual(config.Clients, Clients{}) {
+		return nil, microerror.Maskf(invalidConfigError, "config.Clients must not be empty")
+	}
```

Comment:
I sneaked in checking the Clients dependency. Everything else is removing `DefaultConfig`.

Reply:
We had this already somewhere. Anyway, I think a structure like `Clients` should provide its own `Validate` method in the long run and then we simply call this here.

Reply:
That is a nice approach. I don't have time to do it now but we can improve later.

Reply:
Yeah yeah, for sure. All good. Just saying.